### PR TITLE
Reporting days off

### PR DESCRIPTION
**Pull request by andrzejderylo**
Added reporting days off to nisse:
- added integration with google oauth and google calendar
- added table to keep reported days off
- removed non necessary token tables & endpoints
- started refactoring SlackCommandService to less complex structure.

### CODE REVIEW

File: nisse/routes/__init__.py [hunk 1]
```diff
@@ -5,26 +5,18 @@
...
+def configure_oauth(app: Flask):
+    app.add_url_rule('/google/authorize', 'authorize', google_authorize)    
+    app.add_url_rule('/google/nisseoauthcallback', 'nisseoauthcallback', google_nisseoauthcallback)
+    app.add_url_rule('/google/revoke', 'revoke', google_revoke)    
 
 def configure_api(api: Api):
     # Slack endpoints
     api.add_resource(SlackCommand, '/slack/command', methods=['POST'])
-    api.add_resource(SlackDialogSubmission, '/slack/dialog/submission', methods=['POST'])
+    api.add_resource(SlackDialogSubmission, '/slack/dialog/submission', methods=['POST'])
```
Comment (mjoniak) [inline nisse/routes/__init__.py]: Not much happened here :-)

File: nisse/routes/slack/command_handlers/vacation_command_handler.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                raise ValidationError('Vacation must not start within other vacation. Conflicting vacation: {0} to {1}'.format(vacation.start_date.date(), vacation.end_date.date()), ['start_date'])
+            if vacation.start_date <= end_date and end_date <= vacation.end_date:
+                raise ValidationError('Vacation must not end within other vacation. Conflicting vacation: {0} to {1}'.format(vacation.start_date.date(), vacation.end_date.date()), ['end_date'])
+
+        self.vacation_service.insert_user_vacation(user.user_id, start_date, end_date, payload.submission.reason)
+        self.calendar_service.report_free_day(payload.user.name, user.username, start_date, end_date, payload.submission.reason)
+        self.send_message_to_client(payload.user.id, "Reported vacation from {0} to {1}. Reason: {2}".format(start_date.date(), end_date.date(), payload.submission.reason))
+
+    def create_dialog(self, command_body, argument, action) -> Dialog:
+        nowdate = datetime.now().date() + timedelta(days=1)
```
Comment (mjoniak) [inline nisse/routes/slack/command_handlers/vacation_command_handler.py]: nowdate? i think this is actually tomorrow?

File: nisse/routes/slack/slack_command.py [hunk 3]
```diff
@@ -43,11 +46,13 @@ def post(self):
         arguments = params[1:]
 
         try:
-            result = self.dispatcher.get(action, self.handle_other)(command_body, arguments, action)
+            result = self.dispatcher.get(action, self.handle_other)(
```
Comment (mjoniak) [inline nisse/routes/slack/slack_command.py]: Small stylistic issue, but I would separate it like this:

callback = self.dispatcher.get(action, self.handle_other)
result = callback(command_body, arguments, action)

same number of lines, more readable

File: nisse/routes/slack/slack_interactive_message.py [hunk 4]
```diff
@@ -33,5 +41,26 @@ def post(self):
 
         else:
             payload: Payload = result.data
-            result = payload.handle(self.slack_command_service)
+
+            #this one is for future refactor of slackcommandservice since it is growing bigger and bigger and will eventually turn into god class
```
Comment (mjoniak) [inline nisse/routes/slack/slack_interactive_message.py:45]: :O We must stop this!

File: nisse/services/google_calendar_service.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        free_days_result = self.service.events() \
+            .list(calendarId=self.google_vacation_calendar_id, timeMin=from_date_utc, timeMax=to_date_utc) \
+            .execute()
+        return free_days_result.get('items', [])
+
+    def report_free_day(self, slack_user_name:str, user_email: str, from_date: date, to_date: date, reason:str):                
+        body = {
+            'summary': self.calendar_title_format.format(slack_user_name),
+            'start': {
+                "date": self._format_google_date(from_date),
```
Comment (mjoniak) [inline nisse/services/google_calendar_service.py]: I suggest to keep one style of quotes in whole document (" or ')

https://www.python.org/dev/peps/pep-0008/#string-quotes

**Review by mjoniak — CHANGES_REQUESTED** [review]

File: nisse/services/google_calendar_service.py (hunk 5 above)
Comment (andrzejderylo) [reply nisse/services/google_calendar_service.py]: 👍

File: nisse/routes/slack/slack_command.py (hunk 3 above)
Comment (andrzejderylo) [reply nisse/routes/slack/slack_command.py]: :+1:

File: nisse/routes/__init__.py (hunk 1 above)
Comment (andrzejderylo) [reply nisse/routes/__init__.py]: not really :)

File: nisse/routes/slack/command_handlers/vacation_command_handler.py (hunk 2 above)
Comment (andrzejderylo) [reply nisse/routes/slack/command_handlers/vacation_command_handler.py]: :+1:

**Review by wgebis — APPROVED** [review]

**Review by mjoniak — APPROVED** [review]

**Pull request merged**